docs/reference/hooks.rst

```diff
@@ -2,3 +2,107 @@
 
 Hooks
 =====
+
+A hook is an executable file that runs within a snap’s confined environment when
+a certain action occurs.
+
+Common scenarios requiring hooks include:
+
+- Notifying a snap that something has happened. For example, if a snap has been
+  upgraded, the snap may need to trigger a scripted migration process to port an
+  old data format to the new one.
+
+- Notifying a snap that a specific operation is in progress. For example, a snap may
+  need to know when a specific interface connects or disconnects.
+
+Transactions and rollback
+-------------------------
+
+Snapd executes a hook as a single transaction, where a transaction holds all the
+configuration changes made by the hook. These changes are atomic and are only
+visible to the running system after the hook finishes executing.
+
+This allows changes to be rolled back if errors occur during the execution of a hook.
+This happens if a non-zero value is returned with either the configure or
+default-configure hooks or if an error occurs with any hook involved with
+an `interface auto-connection <https://snapcraft.io/docs/auto-connection-mechanism>`_.
+
+Environment
+-----------
+
+If the hook executable is a shell script, it should assume a POSIX-compliant shell
+environment for its execution.
+
+If your script needs a specific interpreter, such as Bash or Python, it needs to be
+explicitly declared within the script’s hashbang header (``#!/bin/bash``, for example)
```

Comment:
> hashbang

Is this preferred over shebang?

Reply:
Good point, shebang seems to be more common.

Reply:
https://github.com/canonical/snapcraft/pull/5463/commits/5454586af37a127d5d3ef0bbc856bb618916b60b